src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java

```diff
@@ -35,4 +36,13 @@ public List<Book> findAll() {
             throw new EntityNotFoundException("Can't get all books from DB", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try {

```

Comment:
I am sorry, it was my mistake! 
Your first changes were right! 
Could you please return to try-with-resources?